securedrop_client/queue.py

```diff
@@ -28,6 +28,35 @@
 logger = logging.getLogger(__name__)
 
 
+class RunnablePriorityQueue(PriorityQueue):
+    """
+    Wrapper class around PriorityQueue that emits a signal when message or reply
+    download jobs are enqueued or dequeued.
+    """
+
+    def __init__(self, *args: Any, queue_updated_signal: pyqtSignal = None, **kwargs: Any):
+        self.queue_updated_signal = queue_updated_signal
+        super().__init__(*args, **kwargs)
+
+    def get(self, *args: Any, **kwargs: Any) -> Tuple[int, QueueJob]:
+        item = super().get(*args, **kwargs)
+        if self.queue_updated_signal:
+            self.queue_updated_signal.emit(self._get_num_message_or_reply_download_jobs())
+        return item
+
+    def put(self, *args: Any, **kwargs: Any) -> None:
+        item = super().put(*args, **kwargs)
+        if self.queue_updated_signal:
+            self.queue_updated_signal.emit(self._get_num_message_or_reply_download_jobs())
+        return item
+
+    def _get_num_message_or_reply_download_jobs(self) -> int:
+        message_and_reply_download_jobs = list(
+            filter(lambda job: type(job[1]) in (MessageDownloadJob, ReplyDownloadJob), self.queue)
```

Comment:
might be worth adding a comment to explain why you are doing `job[1]`? It's not clear to me here what the type of self.queue is